### ISSUE TITLE
State machine for Payments

### ISSUE THREAD

**Issue by marksteward** [emfcamp/Website#176]
It's getting a bit complicated to keep track of. Put a list of states and transitions in the model, validate the transitions, and only do operations when a change has happened.

**Comment by marksteward** [comment]
Not actually a massive task, but we need to be careful about the differences between payment providers.

**Comment by SamLR** [comment]
Payments have a state field on them but I don't think we've defined a formal state machine for them, do we want to implement that?

**Comment by marksteward** [comment]
Yep, it's not urgent but it's a nice thing to formalise it and helps with testing. I think you'll probably find it throws up one or two missing states for payments, too.

Just something simple like in `models/cfp.py`

**Comment by marksteward** [comment]
Adding good-first-issue as it's doing this for one payment provider is a pretty manageable slice.
